### PR TITLE
FEATURE: Support category, float, group and tag input types

### PR DESCRIPTION
**Pull request by OsamaSayegh**
Continue from https://github.com/discourse/discourse/pull/25673 and https://github.com/discourse/discourse/pull/25811.

This PR adds support for category, float, group and tag types for schema theme settings.

### CODE REVIEW

**Comment by mcwumbly** [comment]
Linter wants those imports sorted

```
$ eslint ./app/assets/javascripts --cache

/__w/discourse/discourse/app/assets/javascripts/admin/addon/components/schema-theme-setting/types/category.gjs
  1:1  error  Run autofix to sort these imports!  simple-import-sort/imports

/__w/discourse/discourse/app/assets/javascripts/admin/addon/components/schema-theme-setting/types/group.gjs
  1:1  error  Run autofix to sort these imports!  simple-import-sort/imports

/__w/discourse/discourse/app/assets/javascripts/admin/addon/components/schema-theme-setting/types/tag.gjs
  1:1  error  Run autofix to sort these imports!  simple-import-sort/imports

/__w/discourse/discourse/app/assets/javascripts/discourse/tests/integration/components/admin-schema-theme-setting/editor-test.gjs
  1:1  error  Run autofix to sort these imports!  simple-import-sort/imports

✖ 4 problems (4 errors, 0 warnings)
  4 errors and 0 warnings potentially fixable with the `--fix` option.
```

File: app/assets/javascripts/admin/addon/components/schema-theme-setting/types/category.gjs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { action } from "@ember/object";
+import CategoryChooser from "select-kit/components/category-chooser";
+
+export default class SchemaThemeSettingTypeCategory extends Component {
+  @tracked value;
+
+  constructor() {
+    super(...arguments);
+    this.value = this.args.value;
+  }
```
Comment (tgxworld) [inline app/assets/javascripts/admin/addon/components/schema-theme-setting/types/category.gjs:13]:
```suggestion
  @tracked value = this.args.value;
```

I believe you can just do this.

File: app/assets/javascripts/admin/addon/components/schema-theme-setting/types/group.gjs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  @tracked value;
+  @tracked groups;
+
+  constructor() {
+    super(...arguments);
+    this.value = this.args.value;
+    Group.findAll().then((groups) => {
+      this.groups = groups;
+    });
+  }
```
Comment (tgxworld) [inline app/assets/javascripts/admin/addon/components/schema-theme-setting/types/group.gjs:17]:
```suggestion
  @tracked value = this.args.value;

  @tracked groups = Group.findAll().then((groups) => {
    this.groups = groups
  });
```

File: app/assets/javascripts/admin/addon/components/schema-theme-setting/types/tag.gjs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { action } from "@ember/object";
+import TagChooser from "select-kit/components/tag-chooser";
+
+export default class SchemaThemeSettingTypeTag extends Component {
+  @tracked value;
+
+  constructor() {
+    super(...arguments);
+    this.value = this.args.value;
+  }
```
Comment (tgxworld) [inline app/assets/javascripts/admin/addon/components/schema-theme-setting/types/tag.gjs:13]:
```suggestion
  @tracked value = this.args.value;
```

File: app/assets/javascripts/admin/addon/components/schema-theme-setting/types/float.gjs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { action } from "@ember/object";
+
+export default class SchemaThemeSettingTypeFloat extends Component {
+  @action
+  onInput(event) {
+    this.args.onChange(parseFloat(event.currentTarget.value));
+  }
+
+  <template>
+    <Input @value={{@value}} {{on "input" this.onInput}} @type="number" />
```
Comment (tgxworld) [inline app/assets/javascripts/admin/addon/components/schema-theme-setting/types/float.gjs:13]: Do you think we should add a `step` attribute here of perhaps `0.1`?

File: app/assets/javascripts/admin/addon/components/schema-theme-setting/types/group.gjs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  @action
+  onInput(newVal) {
+    this.value = newVal;
+    this.args.onChange(newVal);
+  }
+
+  <template>
+    <GroupChooser
```
Comment (tgxworld) [inline app/assets/javascripts/admin/addon/components/schema-theme-setting/types/group.gjs:26]: This seems to be allowing multiple groups to be chosen but we don't support that on the backend yet. Only 1 group id field

**Review by tgxworld — APPROVED** [review]

**Pull request merged**